File: test_api.js
const https = require('https');

const options = {
  hostname: 'sync-learn-updated.onrender.com',
  port: 443,
  path: '/api/mcq/tests',
  method: 'GET',
  headers: {
    'Origin': 'https://synclearn-osd4vkd4f-shubhamm27p.vercel.app'
  }
};

const req = https.request(options, (res) => {
  console.log(`STATUS: ${res.statusCode}`);
  console.log(`HEADERS: ${JSON.stringify(res.headers)}`);
  res.setEncoding('utf8');
  let body = '';
  res.on('data', (chunk) => {
    body += chunk;
  });
  res.on('end', () => {
    console.log(`BODY: ${body}`);
  });
});

req.on('error', (e) => {
  console.error(`problem with request: ${e.message}`);
});

req.end();

File: test_api.py
import requests
try:
    headers = {"Origin": "https://synclearn-osd4vkd4f-shubhamm27p.vercel.app"}
    res = requests.options("https://sync-learn-updated.onrender.com/api/mcq/tests", headers=headers, timeout=10)
    print("OPTIONS Status:", res.status_code)
    print("OPTIONS Headers:", res.headers)
    
    res2 = requests.get("https://sync-learn-updated.onrender.com/api/mcq/tests", headers=headers, timeout=10)
    print("GET Status:", res2.status_code)
    if res2.status_code == 500:
        print("Response:", res2.text)
except Exception as e:
    print(e)
